### PR TITLE
39 legacy connection handler

### LINKED ISSUE #39: `legacy_connection_handling` setter was deprecated in 7.0 and removed in 7.1
```
An error occurred while loading ./spec/model/store_spec.rb.
Failure/Error: require File.expand_path('../config/environment', __dir__)


ArgumentError:
  The `legacy_connection_handling` setter was deprecated in 7.0 and removed in 7.1, but is still defined in your configuration. Please remove this call as it no longer has any effect."
# /home/john/.rvm/gems/ruby-3.2.2/gems/activerecord-7.1.3/lib/active_record.rb:246:in `legacy_connection_handling='
# /home/john/.rvm/gems/ruby-3.2.2/gems/grape-app-0.11.1/lib/grape/app/initializers/pre.rb:47:in `<top (required)>'
# /home/john/.rvm/gems/ruby-3.2.2/gems/zeitwerk-2.6.12/lib/zeitwerk/kernel.rb:38:in `require'
# /home/john/.rvm/gems/ruby-3.2.2/gems/grape-app-0.11.1/lib/grape/app.rb:31:in `init!'
# ./config/environment.rb:6:in `<top (required)>'
```

### PR DESCRIPTION
**Pull request by jpmedforth**

### CODE REVIEW

File: lib/grape/app/initializers/pre.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@
 ActiveSupport::KeyGenerator.hash_digest_class = OpenSSL::Digest::SHA256 if ActiveSupport::KeyGenerator.respond_to?(:hash_digest_class=)
 ActiveSupport::MessageEncryptor.use_authenticated_message_encryption = true
 ActiveSupport::IsolatedExecutionState.isolation_level = :thread if defined?(ActiveSupport::IsolatedExecutionState)
-Digest::UUID.use_rfc4122_namespaced_uuids = true if Digest::UUID.respond_to?(:use_rfc4122_namespaced_uuids=)
+# Digest::UUID.use_rfc4122_namespaced_uuids = true if Digest::UUID.respond_to?(:use_rfc4122_namespaced_uuids=)
```
Comment (jpmedforth) [inline lib/grape/app/initializers/pre.rb]: I'm not sure about this one. Checking....

Comment (dim) [reply lib/grape/app/initializers/pre.rb]: It's fine, `activesupport` is using `SecureRandom.uuid` nowadays, this is legacy from when it used `Digest::UUID` directly. This changed in 7.1, but OK to keep as we support older `activesupport` versions too

File: lib/grape/app/initializers/pre.rb [hunk 2]
```diff
@@ -43,12 +43,6 @@ def call(exception, *)
     ActiveRecord::Base.default_timezone = :utc
   end
 
-  if ActiveRecord.respond_to?(:legacy_connection_handling=)
-    ActiveRecord.legacy_connection_handling = false
-  else
```
Comment (dim) [inline lib/grape/app/initializers/pre.rb:48]: should this be an `elsif`. same here, we should support versions older than `7.1`

File: spec/grape/app_spec.rb [hunk 3]
```diff
@@ -43,14 +43,13 @@ def app
     if ActiveSupport::VERSION::MAJOR >= 7
       expect(ActiveSupport::KeyGenerator.hash_digest_class).to be(OpenSSL::Digest::SHA256)
       expect(ActiveSupport::IsolatedExecutionState.isolation_level).to be(:thread)
-      expect(Digest::UUID.use_rfc4122_namespaced_uuids).to be(true)
+      # expect(Digest::UUID.use_rfc4122_namespaced_uuids).to be(true)
```
Comment (dim) [inline spec/grape/app_spec.rb]: remove the line completely

File: lib/grape/app/initializers/pre.rb (hunk 1 above)
Comment (dim) [reply lib/grape/app/initializers/pre.rb]: just add a comment saying, this is legacy, removed with Rails 7.1 so we know in the future

**Review by dim — CHANGES_REQUESTED** [review]

File: lib/grape/app/initializers/pre.rb [hunk 4]
```diff
@@ -43,6 +44,15 @@ def call(exception, *)
     ActiveRecord::Base.default_timezone = :utc
   end
 
+  # legacy_connection_handling was deprecated in ActiveRecord 7
+  if ActiveRecord::VERSION::MAJOR < 7
```
Comment (dim) [inline lib/grape/app/initializers/pre.rb:48]: why not just `elsif ActiveRecord::Base.respond_to?(:legacy_connection_handling=)`. seems more consistent

Comment (jpmedforth) [reply lib/grape/app/initializers/pre.rb:48]: So it's still in the code. 
https://github.com/rails/rails/blob/v7.1.3/activerecord/lib/active_record.rb#L246

So `respond_to` still returns true, then it raises an error.  I thought best not to try at all if the version was too high.

**Pull request merged**